### PR TITLE
doc/rados: rewrite mclock-config-ref

### PR DESCRIPTION
**Pull request by zdover23**
This PR rewrites the material provided in
PR#40531, improving its elegance and
readability.

This PR rewrites the first 48 percent of the
material provided in PR#40531.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/rados/configuration/mclock-config-ref.rst [hunk 1]
```diff
@@ -4,19 +4,20 @@
...
 
-Based on the profile specified, the OSD determines and applies the lower level
-mclock and Ceph parameters. The parameters applied by the mclock profile
-enables tuning of the QoS between client, background recovery/backfill ops and
-other internal entities (aka best effort clients) within Ceph that generate IOPS
-(for e.g. scrubs, snap trim, PG deletions etc.).
+Using the settings in the specified profile, the OSD determines and applies the
+lower-level mclock and Ceph parameters. The parameters applied by the mclock
+profile make it possible to tune the QoS between client, background
+recovery/backfill ops, and other internal entities (for example, best effort
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:18]: here "internal entities" implies scrubs, snap trim, and PG deletions.

how about something like "QoS between client I/O, recovery/backfill operations, and other background operations (for example scrub, snap trim, and PG deletion). These background activities are considered best effort internal clients of Ceph."?

File: doc/rados/configuration/mclock-config-ref.rst [hunk 2]
```diff
@@ -29,16 +30,18 @@ Ceph cluster enables the throttling of the operations(IOPS) belonging to
...
 osd subop)”*.
 
-The purpose of the mclock profile is to help with the following,
+The purpose of the mclock profile is to help with the following:
 
-- Based on the capacity limits provided and the mclock profile selected,
-  determine the low level mclock resource control parameters,
-- Transparently apply the lower level mclock resource control parameters and
-  some Ceph configuration parameters depending on the profile.
+- the mclock profile uses the capacity limits provided within it and the mclock
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:35]: - it uses the capacity limits and the mclock profile selected by the user to determine the low-level mclock resource control parameters

File: doc/rados/configuration/mclock-config-ref.rst [hunk 3]
```diff
@@ -29,16 +30,18 @@ Ceph cluster enables the throttling of the operations(IOPS) belonging to
...
+  profile that you select to determine the low-level mclock resource control
+  parameters
+- the mclock profile transparently appies the lower-level mclock resource
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:38]: s/appies/applies/

- depending on the profile, lower-level mclock resource control parameters and some Ceph-configuration parameters are transparently applied.

File: doc/rados/configuration/mclock-config-ref.rst [hunk 4]
```diff
@@ -128,44 +130,44 @@ options will be disabled,
 - ``osd_snap_trim_sleep_hybrid``
 
 The above sleep options are disabled to ensure that mclock scheduler is able
-determine when to pick the next op from its operation queue and transfer
-it to the operation sequencer. This results in the desired QoS to be provided
-across all its clients.
+determine when to pick the next op from its operation queue and transfer it to
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:133]: to determine

File: doc/rados/configuration/mclock-config-ref.rst [hunk 5]
```diff
@@ -128,44 +130,44 @@ options will be disabled,
...
-Any existing benchmarking tool may be employed for this purpose and the
-following steps employs the *Ceph Benchmarking Tool* (aka cbt_). Regardless of
-the tool used, the steps described below remain the same.
+Any existing benchmarking tool can be used for this purpose. The following
+steps use the *Ceph Benchmarking Tool* (cbt_). Regardless of the tool
+used, the steps described below remain the same.
 
 As already described in the `OSD Config Reference`_ section, the number of
-shards and the bluestore throttle parameters have an impact on the mclock op
+shards and the bluestore-throttle parameters have an impact on the mclock op
```
Comment (neha-ojha) [inline doc/rados/configuration/mclock-config-ref.rst:159]: bluestore's throttle parameters

File: doc/rados/configuration/mclock-config-ref.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:38]: note to future me: s/appies/applies/

File: doc/rados/configuration/mclock-config-ref.rst [hunk 6]
```diff
@@ -4,19 +4,20 @@
...
 .. index:: mclock; configuration
 
-This dmclock related configuration parameters can be applied using mclock
-profiles. The purpose of the mclock profiles is to mask the low level details
-from a user trying to configure mclock for a specific purpose. At a high level,
-for each OSD in the enviroment the steps involve,
+Dmclock-related configuration parameters can be applied by using mclock
+profiles. Mclock profiles mask the low level details from users, making it
+easier for them to configure mclock. These dmclock-related configuration
+parameters do the following:
```
[the remark this answers is not in the document]
Comment (sseshasa) [reply doc/rados/configuration/mclock-config-ref.rst:10]: Yes, this changes the intent. May be the original content was not clear enough. 

Mclock uses the input parameters like the total OSD capacity and the mclock profile provided by the user and then determines the low-level mclock and Ceph parameters to apply.

File: doc/rados/configuration/mclock-config-ref.rst [hunk 7]
```diff
@@ -29,16 +30,18 @@ Ceph cluster enables the throttling of the operations(IOPS) belonging to
...
+  parameters
+- the mclock profile transparently appies the lower-level mclock resource
+  control parameters and, depending on the profile, some Ceph-configuration
+  parameters.
 
-The lower level mclock resource control parameters are the *reservation, weight
-and limit* that provide control of the resource shares as already described in
-the `OSD Config Reference`_ section.
+The low-level mclock resource control parameters are the *reservation*,
+*limitation*, and *weight* that provide control of the resource shares, as
```
Comment (sseshasa) [inline doc/rados/configuration/mclock-config-ref.rst:43]: The code base uses "limit". Therefore, replace "limitation" with "limit".

File: doc/rados/configuration/mclock-config-ref.rst [hunk 8]
```diff
@@ -48,29 +51,28 @@ mClock Profile Types
...
-  Ceph configuration options.
+- **Custom**: This profile gives users complete control over all mclock and
+  Ceph configuration parameters. Using this profile is not recommended without
+  a deep understanding of mclock and related Ceph-configuration options.
 
-.. note:: Across the built-in profiles, for internal clients of mclock (for e.g.
-          scrub, snap trim etc.) are given slightly lower reservations but
-          higher weight and no limit. This is to ensure that these operations
+.. note:: Across the built-in profiles, internal clients of mclock (for example
+          "scrub" and "snap trim") are given slightly lower reservations, but
```
Comment (sseshasa) [inline doc/rados/configuration/mclock-config-ref.rst:74]: "scrub",  "snap trim" and "pg deletion"

File: doc/rados/configuration/mclock-config-ref.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:10]: I have removed the incorrect sentence and put a short list very similar to Sridhar's suggestion in its place.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:18]: This has been added.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:38]: Done.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 4 above)
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:133]: Done.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 5 above)
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:159]: Done.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 7 above)
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:43]: Done.

File: doc/rados/configuration/mclock-config-ref.rst (hunk 8 above)
Comment (zdover23) [reply doc/rados/configuration/mclock-config-ref.rst:74]: Done.

**Review by neha-ojha — APPROVED** [review]

**Review by sseshasa — APPROVED** [review]

**Pull request merged**